main/main_timer_sync.cpp

```diff
@@ -177,9 +197,37 @@ MainFrameTime MainTimerSync::advance_checked(float p_frame_slice, int p_iteratio
 	// last clamping: make sure time_accum is between 0 and p_frame_slice for consistency between physics and idle
 	ret.clamp_idle(idle_minus_accum, idle_minus_accum + p_frame_slice);
 
+	// all the operations above may have turned ret.idle_step negative or zero, keep a minimal value
+	if (ret.idle_step < min_output_step) {
+		ret.idle_step = min_output_step;
+	}
+
 	// restore time_accum
 	time_accum = ret.idle_step - idle_minus_accum;
 
+	// forcing ret.idle_step to be positive may trigger a violation of the
+	// promise that time_accum is between 0 and p_frame_slice
+#ifdef DEBUG_ENABLED
+	if (time_accum < -1E-7) {
+		WARN_PRINT_ONCE("time_accum negative");
+	}
+#endif
+
+	if (time_accum > p_frame_slice) {
+		const int extra_physics_steps = floor(time_accum * p_iterations_per_second);
+		time_accum -= extra_physics_steps * p_frame_slice;
+		ret.physics_steps += extra_physics_steps;
+	}
+
+#ifdef DEBUG_ENABLED
+	if (time_accum < -1E-7) {
+		WARN_PRINT_ONCE("time_accum negative");
```

Comment:
```suggestion
		WARN_PRINT_ONCE("`time_accum` is negative. This could hint at an engine bug or system timer misconfiguration.");
```

Reply:
Done, squashed and rebased to current master (adapting to name changes there).